docs/conflict_resolution.md

```diff
@@ -0,0 +1,82 @@
+## Conflict Resolution 
+
+### Steps to support Conflict Resolution in your app:
+
+- Create a class, let say `UserConflictResolutionHandler` which extends ConfliceResolutionInterface provided by the library.
+- Override the `resolveConflict()` method of the interface in your class.
+- You get the **`server state`** and the **`client state`** data associated with that mutation in which conflict occurred.
+- You also get the **`operation type`** of the conflicted mutation. Run a switch case on the operation type to detect which type of mutation is it in which conflict occured and accordingly create an object of that mutation while resolving conflicts.
+- You can resolve conflicts based on your business logic and again make a call to the server.
+
+### Code:
+
+```kotlin
+/*
+UserConflictResolutionHandler extends ConflictResolutionInterface.
+Here the user provides the custom implementation of resolving conflicts.
+ */
+class UserConflictResolutionHandler(val context: Context) : ConflictResolutionInterface {
+    val TAG = javaClass.simpleName
+
+    /*User get the server state and the client state.
+     This function resolves the conflicts based on the user business logic.
+     */
+    override fun resolveConflict(
+        serverState: Map<String, Any>,
+        clientState: Map<String, Any>,
+        operationType: String
+    ) {
+        /*Version based approach of Conflict Resolution, for instance.
+        */
+        val serverMap = serverState
+        val containsVersion = serverMap.containsKey("version")
+
+        if (containsVersion) {
```

Comment:
No need for this detailed example, just mention that the user's can resolve conflicts and make a call to the server.

We can have this complex resolution in our sample app.